src/components/dashboard/dashboard.service.ts

```diff
@@ -20,10 +20,11 @@ export class DashboardService implements OnDestroy {
     dimensions$ = new BehaviorSubject<DashboardDimensions>({});
     height$: Observable<number> = this.dimensions$.pipe(tick(), map(dimensions => dimensions.height), distinctUntilChanged());
     placeholder$ = new BehaviorSubject<DashboardPlaceholder>({ visible: false, x: 0, y: 0, width: 0, height: 0 });
-    layout$ = new Subject<DashboardLayoutData[]>();
+    layout$ = new BehaviorSubject<DashboardLayoutData[]>([]);
     stacked$ = new BehaviorSubject<boolean>(false);
     isDragging$ = new BehaviorSubject<DashboardWidgetComponent>(null);
     isGrabbing$ = new BehaviorSubject<DashboardWidgetComponent>(null);
+    layoutChange$ = new Subject<DashboardLayoutData[]>();
```

Comment:
I think the name of this is somewhat misleading, should be something like `dragComplete` or `userLayoutChange`. Also since it's behaving as an event, it should be an EventEmitter.

Reply:
Sorry, keep this as an event, I've been corrected elsewhere on this.